sweep/txgenerator.go

```diff
@@ -75,96 +82,74 @@ func generateInputPartitionings(sweepableInputs []input.Input,
 	// Select blocks of inputs up to the configured maximum number.
 	var sets []inputSet
 	for len(sweepableInputs) > 0 {
-		// Get the maximum number of inputs from sweepableInputs that
-		// we can use to create a positive yielding set from.
-		count, outputValue := getPositiveYieldInputs(
-			sweepableInputs, maxInputsPerTx, feePerKW,
+		// Start building a set of positive-yield tx inputs under the
+		// condition that the tx will be published with the specified
+		// fee rate.
+		txInputs := newTxInputSet(feePerKW)
+
+		// From the set of sweepable inputs, keep adding inputs to the
+		// input set until the tx output value no longer goes up or the
+		// maximum number of inputs is reached.
+		addPositiveYieldInputs(
+			txInputs, sweepableInputs, maxInputsPerTx,
 		)
 
-		// If there are no positive yield inputs left, we can stop
-		// here.
-		if count == 0 {
+		// If there are no positive yield inputs, we can stop here.
+		inputCount := len(txInputs.inputs)
+		if inputCount == 0 {
 			return sets, nil
 		}
 
 		// If the output value of this block of inputs does not reach
 		// the dust limit, stop sweeping. Because of the sorting,
 		// continuing with the remaining inputs will only lead to sets
 		// with a even lower output value.
-		if outputValue < dustLimit {
+		if txInputs.outputValue < dustLimit {
 			log.Debugf("Set value %v below dust limit of %v",
-				outputValue, dustLimit)
+				txInputs.outputValue, dustLimit)
 			return sets, nil
 		}
 
 		log.Infof("Candidate sweep set of size=%v, has yield=%v",
-			count, outputValue)
+			inputCount, txInputs.outputValue)
 
-		sets = append(sets, sweepableInputs[:count])
-		sweepableInputs = sweepableInputs[count:]
+		sweepSet := make([]input.Input, len(txInputs.inputs))
+		copy(sweepSet, txInputs.inputs)
```

Comment:
why copy here rather than slicing as earlier?

Reply:
A suggestion from the linter. I like it because it is less code. Apparently this is the way to do covariance.

Reply:
Looks like it is more code tho? Anyway, no big deal, linter is always right.

Reply:
Oh, you mean even earlier. There was an intermediate revision that had a for loop to make the copy instead of calling `copy`. That is what the linter suggested.

But after that some types changed and it is again possible to slice. Fixed